### PR TITLE
PreferenceStore migration

### PR DESCRIPTION
**Pull request by rvillablanca**
#### Summary

Migration of `PreferenceStore` to return plain errors.

Fixes https://github.com/mattermost/mattermost-server/issues/15010

### CODE REVIEW

File: app/preference.go [hunk 1]
```diff
@@ -12,20 +13,18 @@ import (
 func (a *App) GetPreferencesForUser(userId string) (model.Preferences, *model.AppError) {
 	preferences, err := a.Srv().Store.Preference().GetAll(userId)
 	if err != nil {
-		err.StatusCode = http.StatusBadRequest
-		return nil, err
+		return nil, model.NewAppError("GetAlGetPreferencesForUser", "app.preference.get_all.app_error", nil, err.Error(), http.StatusBadRequest)
```
Comment (agnivade) [inline app/preference.go:16]: Shouldn't this be "GetPreferencesForUser"?

File: app/oauth.go [hunk 2]
```diff
@@ -159,8 +159,8 @@ func (a *App) AllowOAuthAppAccessToUser(userId string, authRequest *model.Author
 		Value:    authRequest.Scope,
 	}
 
-	if err = a.Srv().Store.Preference().Save(&model.Preferences{authorizedApp}); err != nil {
-		mlog.Error("error saving store prefrence", mlog.Err(err))
+	if nErr := a.Srv().Store.Preference().Save(&model.Preferences{authorizedApp}); nErr != nil {
+		mlog.Error("error saving store prefrence", mlog.Err(nErr))
```
Comment (agnivade) [inline app/oauth.go:163]: Let's fix the typo while we are here :)

File: app/preference.go (hunk 1 above)
Comment (rvillablanca) [reply app/preference.go:16]: :ok:

File: app/oauth.go (hunk 2 above)
Comment (rvillablanca) [reply app/oauth.go:163]: :ok:

File: app/preference.go [hunk 3]
```diff
@@ -49,8 +47,14 @@ func (a *App) UpdatePreferences(userId string, preferences model.Preferences) *m
 	}
 
 	if err := a.Srv().Store.Preference().Save(&preferences); err != nil {
-		err.StatusCode = http.StatusBadRequest
-		return err
+		var appErr *model.AppError
+		switch {
+		case errors.As(err, &appErr):
+			appErr.StatusCode = http.StatusBadRequest
+			return appErr
```
Comment (jespino) [inline app/preference.go:54]: nit: There's only one case where it returns an `AppError` and is when is calling `IsValid` and the `IsValid` for `Preference` always return a `http.StatusBadRequest` so I think the overwrite of the status here is not needed.

Comment (rvillablanca) [reply app/preference.go:54]: Will remove that overwrite then, thanks!

**Comment by agnivade** [comment]
Needs EE changes again. I am going to close this and send a new PR with EE changes.

**Comment by rvillablanca** [comment]
Sure!

**Pull request closed without merging**